File: Source/Plugins/Core/com.equella.core/src/com/tle/web/searching/SearchMenuContributor.java
/*
 * Copyright 2017 Apereo
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.tle.web.searching;

import java.util.Collections;
import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import com.dytech.edge.web.WebConstants;
import com.tle.core.guice.Bind;
import com.tle.core.security.TLEAclManager;
import com.tle.web.resources.ResourcesService;
import com.tle.web.sections.SectionInfo;
import com.tle.web.sections.render.Label;
import com.tle.web.sections.result.util.KeyLabel;
import com.tle.web.sections.standard.model.HtmlLinkState;
import com.tle.web.sections.standard.model.SimpleBookmark;
import com.tle.web.template.section.MenuContributor;

@Bind
@Singleton
@SuppressWarnings("nls")
public class SearchMenuContributor implements MenuContributor
{
	private static final Label LABEL_KEY = new KeyLabel(ResourcesService.getResourceHelper(SearchMenuContributor.class)
		.key("searching.menu"));
	private static final String ICON_PATH = ResourcesService.getResourceHelper(SearchMenuContributor.class).url(
		"images/menu-icon-search.png");

	@Inject
	private TLEAclManager aclManager;

	@Override
	public List<MenuContribution> getMenuContributions(SectionInfo info)
	{
		if( aclManager.filterNonGrantedPrivileges(WebConstants.SEARCH_PAGE_PRIVILEGE).isEmpty() )
		{
			return Collections.emptyList();
		}

		HtmlLinkState hls = new HtmlLinkState(new SimpleBookmark("searching.do"));
		hls.setLabel(LABEL_KEY);
		MenuContribution mc = new MenuContribution(hls, ICON_PATH, 1, 20);
		return Collections.singletonList(mc);
	}

	@Override
	public void clearCachedData()
	{
		// Nothing is cached
	}
}

File: Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/searching/SearchMenuContributor.scala
/*
 * Licensed to The Apereo Foundation under one or more contributor license
 * agreements. See the NOTICE file distributed with this work for additional
 * information regarding copyright ownership.
 *
 * The Apereo Foundation licenses this file to you under the Apache License,
 * Version 2.0, (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.tle.web.searching

import java.util
import java.util.Collections
import com.dytech.edge.web.WebConstants
import com.tle.core.i18n.CoreStrings
import com.tle.core.security.ACLChecks.hasAcl
import com.tle.web.sections.SectionInfo
import com.tle.web.sections.result.util.KeyLabel
import com.tle.web.sections.standard.model.{HtmlLinkState, SimpleBookmark}
import com.tle.web.settings.UISettings
import com.tle.web.template.RenderNewTemplate
import com.tle.web.template.section.MenuContributor
import com.tle.web.template.section.MenuContributor.MenuContribution

object SearchMenuContributor extends MenuContributor {
  private val LABEL_KEY = new KeyLabel(CoreStrings.key("searching.menu"))
  private val ICON_PATH = CoreStrings.lookup.url("images/menu-icon-search.png")

  override def getMenuContributions(info: SectionInfo): util.List[MenuContribution] = {
    if (!hasAcl(WebConstants.SEARCH_PAGE_PRIVILEGE)) {
      Collections.emptyList()
    } else {
      val uis          = UISettings.getUISettings
      val useNewSearch = uis.newUI.newSearch && RenderNewTemplate.isNewLayout(info)
      val hls = new HtmlLinkState(
        new SimpleBookmark(if (useNewSearch) "page/search" else "searching.do")
      )
      hls.setLabel(SearchMenuContributor.LABEL_KEY)
      val mc = new MenuContributor.MenuContribution(
        hls,
        SearchMenuContributor.ICON_PATH,
        1,
        20,
        "search",
        if (useNewSearch) "/page/search" else null
      )
      Collections.singletonList(mc)
    }
  }

  override def clearCachedData(): Unit = {
    // Nothing is cached
  }
}
